internal/framework/status/updater_test.go

```diff
@@ -399,10 +400,10 @@ var _ = Describe("Updater", func() {
 
 				expectedGc.ResourceVersion = latestGc.ResourceVersion
 
-				Expect(helpers.Diff(expectedGc, latestGc)).To(BeEmpty())
+				Expect(helpers.Diff(expectedGc, latestGc)).ToNot(BeEmpty())
```

Comment:
I think it would be better to check that the gateway class still has the old generation here. The diff between expected and actual could be anything. We could have accidentally named the expected resource the wrong thing and this test would still pass.

Same goes for the other tests.